TestHelper.psm1

```diff
@@ -201,3 +204,221 @@ function Install-ResourceDesigner {
+        2. Copying this module to the PSModules folder.
+        3. Backing up any settings that need to be changed to accurately test
+           a resource.
+        4. Producing a test object containing any parameters that may be used
+           for testing as well as storing the backed up settings.
+    
+    .PARAMETER DSCModuleName
+        The name of the DSC Module containing the resource that the tests will be
+        run on.
+
+    .PARAMETER DSCResourceName
+        The full name of the DSC resource that the tests will be run on. This is
+        usually the name of the folder containing the actual resource MOF file.
+
+    .PARAMETER TestType
+        Specifies the type of tests that are being intialized. It can be:
+        Unit: Initialize for running Unit tests on a DSC resource. Default.
+        Integration: Initialize for running Integration tests on a DSC resource.
+
+    .OUTPUT
+        Returns a test environment object which must be passed to the
+        Restore-TestEnvironment function to allow it to restore the system
+        back to the original state as well as clean up and working/temp files.
+        
+    .EXAMPLE
+        $TestEnvrionment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            
+        This command will initialize the test enviroment for Unit testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.      
+
+    .EXAMPLE
+        $TestEnvrionment = Inialize-TestEnvironment `
+            -DSCModuleName 'xNetworking' `
+            -DSCResourceName 'MSFT_xFirewall'
+            -TestType Integration
+            
+        This command will initialize the test enviroment for Integration testing
+        the MSFT_xFirewall DSC resource in the xNetworking DSC module.    
+#>
+function Initialize-TestEnvironment
+{
+    [OutputType([PSObject])]
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCModuleName,
+        
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DSCResourceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Unit','Integration')]
+        [String] $TestType
+    )
+    
+    Write-Verbose -Verbose (`
+        'Initializing Test Environment for {0} testing of {1} in module {2}.' `
+            -f $TestType,$DSCResourceName,$DSCModuleName)   
+    if ($TestType -eq 'Unit')
+    {
+        [String] $RelativeModulePath = "DSCResources\$DSCResourceName\$DSCResourceName.psm1"
+    }
+    else
+    {
+        [String] $RelativeModulePath = "$DSCModuleName.psd1"
+    }
+
+    # Temp Working Folder - always gets removed on completion
+    # The tests can put anything in here and it will get cleaned up.
+    [String] $WorkingFolder = Join-Path -Path $env:Temp -ChildPath $DSCResourceName
```

Comment:
Can we also create this folder here so that tests can safely assume it exists?
Let's also add some random number to name of this folder so that we don't have any collisions/remove someone's existing data.